libs/datalayout/src/main/java/com/linkedin/openhouse/datalayout/datasource/TablePartitionStats.java

```diff
@@ -0,0 +1,55 @@
+package com.linkedin.openhouse.datalayout.datasource;
+
+import java.util.ArrayList;
+import java.util.List;
+import lombok.Builder;
+import org.apache.spark.api.java.function.MapFunction;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Encoders;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.types.StructType;
+
+/** Data source implementation for table partition statistics. */
+@Builder
+public class TablePartitionStats implements DataSource<PartitionStat> {
+  private final SparkSession spark;
+  private final String tableName;
+
+  /**
+   * Get partition statistics dataset for the table.
+   *
+   * @return Dataset of partition statistics sorted by partition values.
+   */
+  @Override
+  public Dataset<PartitionStat> get() {
+    StructType partitionSchema =
+        spark.sql(String.format("SELECT * FROM %s.partitions", tableName)).schema();
+    try {
+      partitionSchema.apply("partition");
+      return spark
+          .sql(
+              String.format(
+                  "SELECT partition, file_count FROM %s.partitions ORDER BY partition", tableName))
```

Comment:
```suggestion
                  "SELECT partition, file_count FROM %s.partitions", tableName))
```
Not sure ordering is needed. If yes, method should be getOrdered() or something.

Reply:
Removed sorting for now, generator could do sorting later if needed.